pippy/PipelineSchedule.py

```diff
@@ -517,8 +519,12 @@ def compute_loss(self):
 
 
 class PipelineSchedule(ABC):
+    def __init__(self, stage: PipelineStageBase, n_microbatches: int):
```

Comment:
does `n_microbatches` need to be fixed at initialization? Should we allow `step()` to take microbatches of differing lengths?

Reply:
We can. 
The caveat is that while rank 0 can figure out `n_microbatches` from `len(microbatches)`, the middle ranks do not take `microbatches` as input (as changed in this PR). We can let `step` to take `n_microbatches`, but then it also seems redundant for rank 0. 
What's your pref?